.github/workflows/required-tests.yml

```diff
@@ -0,0 +1,211 @@
+name: Required Tests
+
+on:
+    push:
+        branches:
+            - master
+            - gh-actions
+    pull_request:
+        branches:
+            - master
+
+jobs:
+    # "Build" job
+    build:
+      # This job tries to build PKI from src on a fresh docker container.
+      # The docker container is spawned by github itself and we merely just
+      # run the build commands. We then upload the artifact for consumption
+      # by the test jobs + for the public to consume. This job **does not**
+      # upload any build logs as they are visible in the log console itself.
+ 
+      name: Build PKI
+      runs-on: ubuntu-latest
+      container: fedora:${{ matrix.os }}
+      strategy:
+          matrix:
+            os: ['30', '31']
+      steps:
+          - name: Update base image
+            run: |
+                  set -x &&
+                  dnf update -y &&
+                  dnf install -y dnf-plugins-core rpm-build git
+
+          - name: Clone the repository
+            uses: actions/checkout@v2
+
+          - name: Install PKI build deps
+            run: |
+                  # dnf copr enable @pki/master
+                  dnf builddep -y --allowerasing --spec ./pki.spec &&
+                  dnf clean all
```

Comment:
We probably can remove `dnf clean all` since this is just a temporary container. It might shorten the build time a little bit.